### PR TITLE
add enhancements of profile readmes

### PR DESCRIPTION
**Pull request by a1012112796**
follow #23260, do some enhancement to make it better.

- reuse readme file finding and render logic in repo file render function, then all features in repo's readme file like locolize file name, file reference (like using figure in profile repo) and issue/pull reference will be usefull alse. that's same with github.
example:
image

- make profile repo name configable for all user (TIP: you can disable it by entry a not exist repository name :) )
- less `serverError` in profile readme render logic

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -627,6 +627,7 @@ continue = Continue
 cancel = Cancel
 language = Language
 ui = Theme
+profile_repo_name=profile repo name
```
Comment (yardenshoham) [inline options/locale/locale_en-US.ini]:
```suggestion
profile_repo_name = Profile repo name
```

File: routers/web/repo/render/render.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+type FileInfo struct {
+	isTextFile bool
+	isLFSFile  bool
+	fileSize   int64
+	lfsMeta    *lfs.Pointer
+	St         typesniffer.SniffedType
+}
+
+func (f *FileInfo) IsTextFile() bool {
+	return f.isLFSFile
```
Comment (yardenshoham) [inline routers/web/repo/render/render.go]:
```suggestion
	return f.isTextFile
```

**Review by yardenshoham — COMMENTED** [review]
We should also not show private repos but that can be another PR

File: models/user/setting_keys.go [hunk 3]
```diff
@@ -12,4 +12,6 @@ const (
 	UserActivityPubPrivPem = "activitypub.priv_pem"
 	// UserActivityPubPubPem is user's public key
 	UserActivityPubPubPem = "activitypub.pub_pem"
+	// SettingsKeyProfileRepoName is user's profile repo name
+	SettingsKeyProfileRepoName = "profile_repo_name"
```
Comment (lunny) [inline models/user/setting_keys.go]: It's better to use two sections split with a dot symbol.

Comment (a1012112796) [reply models/user/setting_keys.go]:
```suggestion
	SettingsKeyProfileRepoName = "profile.repo_name"
```

How about this?

**Pull request closed without merging**